api/test/trace/propagation/http_text_format_test.cc

```diff
@@ -0,0 +1,209 @@
+#include "opentelemetry/context/context.h"
+#include "opentelemetry/nostd/shared_ptr.h"
+#include "opentelemetry/nostd/span.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/trace/default_span.h"
+#include "opentelemetry/trace/span.h"
+#include "opentelemetry/trace/span_context.h"
+#include "opentelemetry/trace/trace_id.h"
+#include "opentelemetry/trace/tracer.h"
+
+#include <map>
+#include <memory>
+#include <string>
+
+#include <gtest/gtest.h>
+
+#include "opentelemetry/trace/default_span.h"
+#include "opentelemetry/trace/propagation/http_text_format.h"
+#include "opentelemetry/trace/propagation/http_trace_context.h"
+
+using namespace opentelemetry;
+
+static nostd::string_view Getter(const std::map<std::string, std::string> &carrier,
+                                 nostd::string_view trace_type = "traceparent")
+{
+  auto it = carrier.find(std::string(trace_type));
+  if (it != carrier.end())
+  {
+    return nostd::string_view(it->second);
+  }
+  return "";
+}
+
+static void Setter(std::map<std::string, std::string> &carrier,
+                   nostd::string_view trace_type        = "traceparent",
+                   nostd::string_view trace_description = "")
+{
+  carrier[std::string(trace_type)] = std::string(trace_description);
+}
+
+static trace::propagation::HttpTraceContext<std::map<std::string, std::string>> format =
+    trace::propagation::HttpTraceContext<std::map<std::string, std::string>>();
+
+using MapHttpTraceContext =
+    trace::propagation::HttpTraceContext<std::map<std::string, std::string>>;
+TEST(HTTPTextFormatTest, TraceIdBufferGeneration)
+{
+  constexpr uint8_t buf[] = {1, 2, 3, 4, 5, 6, 7, 8, 8, 7, 0xaa, 0xbb, 0xcc, 0xdd, 0xee, 0xff};
+  EXPECT_EQ(MapHttpTraceContext::GenerateTraceIdFromString("01020304050607080807aabbccddeeff"),
+            trace::TraceId(buf));
+}
+
+TEST(HTTPTextFormatTest, SpanIdBufferGeneration)
+{
+  constexpr uint8_t buf[] = {1, 2, 0xaa, 0xbb, 0xcc, 0xdd, 0xee, 0xff};
+  EXPECT_EQ(MapHttpTraceContext::GenerateSpanIdFromString("0102aabbccddeeff"), trace::SpanId(buf));
+}
+
+TEST(HTTPTextFormatTest, TraceFlagsBufferGeneration)
+{
+  EXPECT_EQ(MapHttpTraceContext::GenerateTraceFlagsFromString("00"), trace::TraceFlags());
+}
+
+// The commented out code below are ones that are related to TraceState. Needs to be uncommented
```

Comment:
You could remove the commented-out code, save the code somewhere (e.g. in another branch), and add this code later on once `TraceState` is merged. Same goes for other commented-out code.

Reply:
I'll close this PR then.